pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadRestletResource.java

```diff
@@ -315,12 +317,13 @@ private SuccessResponse uploadSegment(FormDataMultiPart multiPart, boolean enabl
               clientAddress);
 
       // Validate segment
-      new SegmentValidator(_pinotHelixResourceManager, _controllerConf, _executor, _connectionManager,
-          _controllerMetrics).validateSegment(segmentMetadata, tempSegmentDir);
+      SegmentValidatorResponse segmentValidatorResponse =
+          new SegmentValidator(_pinotHelixResourceManager, _controllerConf, _executor, _connectionManager,
```

Comment:
Just curious if we need to allocate a segmentvalidator everytime - or if we can just expose a static method on a util like class.

Reply:
SegmentValidator also depends on other variables like `pinotHelixResourceManager`, `executor`, `connectionManager`, etc, which are not related to the segment itself. And the logic is in Resource class. I think it's best to instantiate one, since the latency of uploading an offline segment is acceptable.

Reply:
I agree that this should be a normal class instead of a util class. This way in the future, it will be easier to add more functionalities, or split the current one into multiple steps (e.g. checkTimeStamp(), checkResourceAllocation() etc.).